arangod/Replication2/Methods.cpp

```diff
@@ -220,8 +221,31 @@ struct ReplicatedLogMethodsDBServer final
 
   auto release(LogId id, LogIndex index) const
       -> futures::Future<Result> override {
-    auto log = vocbase.getReplicatedLogById(id);
-    return log->getParticipant()->release(index);
+    auto stateBase = vocbase.getReplicatedStateById(id);
+    if (!stateBase.ok()) {
+      return std::move(stateBase).result();
+    }
+    if (auto state =
+            std::dynamic_pointer_cast<replicated_state::ReplicatedState<
+                replicated_state::black_hole::BlackHoleState>>(stateBase.get());
+        state != nullptr) {
+      if (auto leaderState = state->getLeader(); leaderState != nullptr) {
+        return leaderState->release(index);
+      } else {
+        return Result(
+            TRI_ERROR_HTTP_FORBIDDEN,
+            fmt::format("/release API is only allowed on leaders; while "
+                        "trying to release state machine {} on {}.",
+                        id, ServerState::instance()->getId()));
```

Comment:
I was under the impression that `/release` can be called on followers as well. Each state machine (be it leader or follower) can call release independently.
Will the `leaderState->release(index)` call propagate to followers?
Not that restricting the `/release` like this would do any harm, since it is only used for testing. I'm just curios to learn why.

Reply:
We could allow it on the followers. But the API is now restricted to the black hole state machine already, just to support the tests. I would have removed it completely if it weren't for the tests; and I think we would write the tests differently today, so we wouldn't need it (i.e. unit tests).